src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        private const string FrameworkName = ".NET Core";
+        private static Architecture? s_processArch = null;
+        private static readonly bool s_is64BitProcess = IntPtr.Size == 8;
+
+        public static string FrameworkDescription
+        {
+            get
+            {
+                return FrameworkName;
+            }
+        }
+
+        public static Architecture ProcessArchitecture
+        {
+            get
+            {
+                if (null == s_processArch)
+                {
+                    if (Architecture.Arm == OSArchitecture)
```

Comment:
You current code tries to guess the processor arch by making fragile assumptions about the valid combinations. They will be broken as more combinations are added.

`GetSystemInfo` gives you the Processor Arch accurately, so you should just call it and avoid the guessing.